File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        Map<Integer,Integer> mp=new HashMap<>();
        for(int i=0;i<n;i++){
            int remaining=target-nums[i];
            if(mp.containsKey(remaining)){
                return new int[]{mp.get(remaining),i};
            }
            mp.put(nums[i],i);
        }   
        return new int[0];
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            int a = nums[i];
            int b = target - a;
            if(mp.find(b)!=mp.end()){
                return {i,mp[b]};
            }
            mp[a]=i;
        }
        return {};
    }
};